plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/IcebergRegisterTableUtil.java

```diff
@@ -0,0 +1,119 @@
+import static io.trino.plugin.base.util.Procedures.checkProcedureArgument;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_FILESYSTEM_ERROR;
+import static io.trino.plugin.iceberg.IcebergErrorCode.ICEBERG_INVALID_METADATA;
+import static io.trino.plugin.iceberg.IcebergUtil.METADATA_FILE_EXTENSION;
+import static io.trino.plugin.iceberg.IcebergUtil.METADATA_FOLDER_NAME;
+import static io.trino.plugin.iceberg.IcebergUtil.parseVersion;
+import static io.trino.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static java.lang.String.format;
+
+public class IcebergRegisterTableUtil
+{
+    private IcebergRegisterTableUtil() {}
+
+    public static String getLatestMetadataLocation(TrinoFileSystem fileSystem, String location)
+    {
+        List<String> latestMetadataLocations = new ArrayList<>();
+        String metadataDirectoryLocation = format("%s/%s", stripTrailingSlash(location, "table_location"), METADATA_FOLDER_NAME);
+        try {
+            int latestMetadataVersion = -1;
+            FileIterator fileIterator = fileSystem.listFiles(metadataDirectoryLocation);
+            while (fileIterator.hasNext()) {
+                FileEntry fileEntry = fileIterator.next();
+                if (fileEntry.path().contains(METADATA_FILE_EXTENSION)) {
+                    OptionalInt version = parseVersion(fileEntry.path());
+                    if (version.isPresent()) {
+                        int versionNumber = version.getAsInt();
+                        if (versionNumber > latestMetadataVersion) {
+                            latestMetadataVersion = versionNumber;
+                            latestMetadataLocations.clear();
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                        else if (versionNumber == latestMetadataVersion) {
+                            latestMetadataLocations.add(fileEntry.path());
+                        }
+                    }
+                }
+            }
+            if (latestMetadataLocations.isEmpty()) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, "No versioned metadata file exists at location: " + metadataDirectoryLocation);
+            }
+            if (latestMetadataLocations.size() > 1) {
+                throw new TrinoException(ICEBERG_INVALID_METADATA, format(
+                        "More than one latest metadata file found at location: %s, latest metadata files are %s",
+                        metadataDirectoryLocation,
+                        latestMetadataLocations));
+            }
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_FILESYSTEM_ERROR, "Failed checking table's location: " + location, e);
+        }
+        return getOnlyElement(latestMetadataLocations);
+    }
+
+    public static String stripTrailingSlash(String value, String name)
+    {
+        checkNotEmpty(value, name);
```

Comment:
remove. let the caller do checks with nice message
here we will just NPE (with an OK message) if `value` is null

Reply:
Removed null check